### PR TITLE
Ensure Json parsers are not removed during build when using kafka client.

### PR DESCRIPTION
**Pull request by andreas-repp**

### CODE REVIEW

**Comment by andreas-eberle** [comment]
@cescoffier: This PR should fix it. However, my local build is still running, so I wasn't able to test it yet.

File: extensions/kafka-client/deployment/src/main/java/io/quarkus/kafka/client/deployment/KafkaProcessor.java [hunk 1]
```diff
@@ -255,4 +258,9 @@ HealthBuildItem addHealthCheck(KafkaBuildTimeConfig buildTimeConfig) {
         return new HealthBuildItem("io.quarkus.kafka.client.health.KafkaHealthCheck",
                 buildTimeConfig.healthEnabled);
     }
+
+    @BuildStep
+    UnremovableBeanBuildItem ensureJsonParserAvailable() {
+        return UnremovableBeanBuildItem.beanTypes(ObjectMapper.class, Jsonb.class);
```
Comment (cescoffier) [inline extensions/kafka-client/deployment/src/main/java/io/quarkus/kafka/client/deployment/KafkaProcessor.java:264]: USe the string version of the class names, so you don't have to pull the dependencies.

File: extensions/kafka-client/deployment/pom.xml [hunk 2]
```diff
@@ -35,6 +35,14 @@
             <groupId>io.quarkus</groupId>
             <artifactId>quarkus-arc-deployment</artifactId>
         </dependency>
+        <dependency>
+            <groupId>io.quarkus</groupId>
```
Comment (cescoffier) [inline extensions/kafka-client/deployment/pom.xml:39]: let's try to avoid pulling these extensions

**Review by cescoffier — DISMISSED** [review]

File: extensions/kafka-client/deployment/src/main/java/io/quarkus/kafka/client/deployment/KafkaProcessor.java (hunk 1 above)
Comment (andreas-eberle) [reply extensions/kafka-client/deployment/src/main/java/io/quarkus/kafka/client/deployment/KafkaProcessor.java:264]: Isn't that risky in the case the classes ever get renamed or moved? Can this be detected by the compiler then?

Comment (andreas-eberle) [reply extensions/kafka-client/deployment/src/main/java/io/quarkus/kafka/client/deployment/KafkaProcessor.java]: @cescoffier: I switched to using the string version as you suggested. When adding the deployment dependencies, the build didn't work anymore anyways. The current version worked for me.

Comment (gsmet) [reply extensions/kafka-client/deployment/src/main/java/io/quarkus/kafka/client/deployment/KafkaProcessor.java]: It's a bit more fragile but it's the only to not end up with a big dependency mess :).

**Comment by andreas-eberle** [comment]
@cescoffier: Can you see what's wrong with the build? Did it simply time out?

**Review by gsmet — APPROVED** [review]

**Pull request merged**